### DISCUSSION TITLE
Mirth administrator menu size / scrolling issues

### DISCUSSION THREAD

**Discussion by ppazos** [nextgenhealthcare/connect#6087] [Feature/Enhancement Ideas]
Hi all,

I've been having these issues for a while with the administrator, now it's just getting annoying for the time I lose trying to see things I can't.

The first issue is some scrollable areas just don't go all the way down, so I can't see, for instance, the complete configuration of source connectors. This is important because I can't change the configurations I don't see.

Second issue is the width of the menu is too short to display long texts so those get clipped. Though I know what I should see, it's annoying for instance when sharing my screen with a colleague and I need to point to an item in the menu that says "Edit Transforme...". It would be better that the width adapts to the longest item in the menu, or the user can control the width to the desired size.

![mirth-administrator-size-problems](https://github.com/nextgenhealthcare/connect/assets/1278556/ac914d26-c7c6-4941-8ac0-c6b39d6be28f)

**Comment by pacmano1** [comment]
You need to add windows fonts to your Linux box if I recall correctly.  But yeah, all that could be better.

**Reply by ppazos** [reply]
Hi @pacmano1 how can I do that?

If that is the solution, I believe the "work area" height might be calculated based on the font height? Isn't there a more dynamic way of calculating the height or maybe leaving some padding at the bottom just in case?

Note if I change to a bigger resolution screen, the scrolling is not a problem, but on my management workstation the resolution is 1366*768 (that's on the screenshot). 

For the menu I don't think the issue is the resolution. In the big monitor the text is still clipped. Maybe putting a handle so the user can control the width of the column, with a min-width value, would be useful. Or by dynamically calculating the width from the longest item in the menu. I can do a little research. What's the GUI framework used in the administrator?

**Reply by pacmano1** [reply]
Someone needs a better monitor.   One second on which font it was.. I forgot.   Gotta fire up a VM to check.

**Reply by pacmano1** [reply]
Before:

![image](https://github.com/nextgenhealthcare/connect/assets/44065187/a4fea851-a55b-4efa-b529-b6f971663719)

After tahoma.ttf install:

![image](https://github.com/nextgenhealthcare/connect/assets/44065187/97a8ff9b-e3e4-4a1f-9015-601a96d24839)

**Reply by ppazos** [reply]
@pacmano1 but how do you physically install the font? I don't have a windows box to grab the font. Is there a font repo somewhere?

On the other hand, should installing a font be a technical requirement for the administrator to work correctly? ¯\_(ツ)_/¯

**Reply by pacmano1** [reply]
I don't work for Nextgen, aka "not my fault"!  lol.

**Reply by jonbartels** [reply]
It's been a while but I think there is a package for that for most distros now. `ttf-mscorefonts-installer` maybe?  https://itsfoss.com/install-microsoft-fonts-ubuntu/

**Reply by ppazos** [reply]
Hey Nextgen hire @pacmano1 so it can be his fault :P

But seriously, if this affects the usage of the administrator, it should be a technical requirement for linux.

**Reply by pacmano1** [reply]
@jonbartels Tahoma not in that package (I think).

You gotta grab the font from "somewhere", like a windows box.   You are responsible for licenses of course.

See https://aur.archlinux.org/packages/ttf-tahoma though.

**Reply by ppazos** [reply]
Ouch, that font is terrible, now everything looks like in the 90s... I think is the space between characters in my previous font made labels more readable. This tahoma font makes text thinner but really difficult to read (I have sight problems).

![Screenshot_2024-02-06_17-30-00](https://github.com/nextgenhealthcare/connect/assets/1278556/eac3b8f2-acc8-493c-8053-1c8a5d6fb10a)

How can I report this as a bug?

**Reply by pacmano1** [reply]
Scroll to the top of this page and go to the issue section.

I doubt you will get traction though.  There are already feature requests to allow a font choice.   And I would SO want a dark theme first, lol.

**Reply by ppazos** [reply]
I don't need the font choice thing, it seems Java is smartly getting a similar font from the OS when tahoma is not there. I just need that, with whatever font is displayed, the menu doesn't get clipped and the scroll doesn't stop before displaying the whole content :)

**Comment by jonbartels** [comment]
https://github.com/search?q=repo%3Anextgenhealthcare%2Fconnect+tahoma&type=issues it's a long standing issue unfortunately.

**Reply by ppazos** [reply]
That's sad...

**Reply by pacmano1** [reply]
Someone get @ppazos a mac!

**Reply by ppazos** [reply]
That would be nice @pacmano1 though I can't spend thousands of $$$ on non-standard adapters :P
